arch/arm/core/elf.c

```diff
@@ -29,6 +68,22 @@ void arch_elf_relocate(elf_rela_t *rel, uintptr_t opaddr, uintptr_t opval)
 		/* Update the absolute address of a load/store instruction */
 		*((uint32_t *)opaddr) = (uint32_t)opval;
 		break;
+	case R_ARM_ARM_THM_CALL:
+		/*
+		 * The branch offset (addend) is encoded into the value at opaddr.
+		 * Start by subtracting the raw value of opaddr added to opval
+		 */
+		opval -= *((uint32_t *)opaddr);
```

Comment:
This is definitely highlighting an issue with the current code, and I think it's time to fix it. 
The source is [this calculation](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/llext/llext.c#L771-L774), which assumes `*op_loc` contains the offset; that is in fact true only for `R_ARM_ABS32` relocs.
Can you fix this in a commit that moves this addend from `llext.c` to the `R_ARM_ABS32` case above? That way you will not need this odd bit of magic.

Reply:
Absolutely :) I was going to create a PR specifically for this, may as well fix it now

Reply:
Thank you